components/post_view/post_list/index.js

```diff
@@ -64,13 +65,17 @@ function makeMapStateToProps() {
             atOldestPost = chunk.oldest;
         }
 
+        let currentUserLastPostId;
         if (postIds) {
             formattedPostIds = preparePostIdsForPostList(state, {postIds, lastViewedAt, indicateNewMessages: true, channelId});
             if (postIds.length) {
                 const latestPostId = memoizedGetLatestPostId(postIds);
                 const latestPost = getPost(state, latestPostId);
                 latestPostTimeStamp = latestPost.create_at;
                 latestAriaLabelFunc = createAriaLabelForPost(state, latestPost);
+                if (showPostHeaderUpdateStatusButton(state)) {
+                    currentUserLastPostId = getLastCurrentUserPostWithHeaderId(state, postIds);
```

Comment:
Alright, so I talked with Eric, Matt, and Dean about this, and we came to the conclusion that this will be very difficult to do with the way the post list is currently structured, specifically because determining the most recent post that fits those requirements is very hard. We have some ideas for solving that in the future for a more general tutorial system, but we think this isn't going to be problem for custom statuses.

We decided to compromise and show the Set a Custom Status on every post that meets the requirements of being made by the current user and having a header. It might be a bit more annoying for the user, but I, personally, think that might make it so users are more likely to click on it.